### PR TITLE
Deprecate `astroquery.utils.commons.send_request`

### LINKED ISSUE #825: Deprecate commons.send_request
Once the last usage of ``commons.send_request`` is removed, see #824, it should be deprecated in favour of ``BaseQuery._request``.

### PR DESCRIPTION
**Pull request by eerovaher**
Closes #825.

### CODE REVIEW

**Review by bsipocz — APPROVED** [review]
Overall it looks good, but am waiting for Adam's input on the changelog before merging. Thanks @eerovaher!

File: CHANGES.rst [hunk 1]
```diff
@@ -41,6 +41,9 @@ Infrastructure, Utility and Other Changes and Additions
   The API for JPLspec's ``lookup_table.find`` function returns a dictionary
   instead of values (for compatibility w/CDMS).  [#2144]
 
+- The function ``astroquery.utils.commots.send_request()`` has been deprecated
```
[the remark this answers is not in the document]
Comment (keflavich) [reply CHANGES.rst:44]: No, I don't think anyone uses it.  I think it's fine to treat this as an internal deprecation and leave it out of the changelog.  If someone files an Issue saying, "Where has `send_request` gone?!", I'll be shocked.

[the remark this answers is not in the document]
Comment (eerovaher) [reply CHANGES.rst]: I have removed the change log entry.

**Comment by bsipocz** [comment]
Thanks @eerovaher!

**Pull request merged**